db/dcrsqlite/apisource.go

```diff
@@ -992,25 +992,40 @@ func (db *WiredDB) GetPoolValAndSizeRange(idx0, idx1 int) ([]float64, []float64)
 	return poolvals, poolsizes
 }
 
-// GetSqliteChartsData fetches the charts data from the sqlite db.
-func (db *WiredDB) GetSqliteChartsData() (map[string]*dbtypes.ChartsData, error) {
-	poolData, err := db.RetrieveAllPoolValAndSize()
-	if err != nil {
-		return nil, err
+// SqliteChartsData fetches the charts data from the sqlite db.
+func (db *WiredDB) SqliteChartsData(data []*dbtypes.ChartsData) (err error) {
+	if len(data) != dbtypes.SqliteChartsCount {
+		data = make([]*dbtypes.ChartsData, dbtypes.SqliteChartsCount)
 	}
 
-	feeData, err := db.RetrieveBlockFeeInfo()
+	feeData := data[dbtypes.FeePerBlock.SqlitePos()]
+	feeData, err = db.RetrieveBlockFeeInfo(feeData)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	var data = map[string]*dbtypes.ChartsData{
-		"ticket-pool-size":  {Time: poolData.Time, SizeF: poolData.SizeF},
-		"ticket-pool-value": {Time: poolData.Time, ValueF: poolData.ValueF},
-		"fee-per-block":     feeData,
+	poolData := data[dbtypes.TicketPoolSize.SqlitePos()]
+	poolValue := data[dbtypes.TicketPoolValue.SqlitePos()]
+
+	var wCopy *dbtypes.ChartsData
+	//  Append the all data to the working Copy.
+	if poolData != nil && poolValue != nil {
+		wCopy = new(dbtypes.ChartsData)
+		*wCopy = *poolData
+		wCopy.ValueF = poolValue.ValueF
```

Comment:
None of the underlying slices' data buffers are copies.  They still share the same memory.

Reply:
I found it to affect the underlying array thus the working copy was the solution.